src/main/java/uk/gov/companieshouse/api/accounts/utility/BindingResultErrorToErrorMapper.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.companieshouse.api.accounts.utility;
+
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.model.validation.Error;
+import uk.gov.companieshouse.api.accounts.validation.ErrorMessageKeys;
+import uk.gov.companieshouse.api.accounts.validation.ErrorType;
+import uk.gov.companieshouse.api.accounts.validation.LocationType;
+
+import java.util.ArrayList;
+
+
+@Component
+public class BindingResultErrorToErrorMapper {
+
+    String location;
+
+    /**
+     * Maps each binding result error to {@link Error} model, and adds to returned {@link Errors}
+     *
+     * @param bindingResult
+     * @param errors
+     * @return
+     */
+    public Errors mapBindingResultErrorsToErrorModel(BindingResult bindingResult, Errors errors) {
+
+        for (Object object : bindingResult.getAllErrors()) {
+
+            if (object instanceof FieldError) {
+                FieldError fieldError = (FieldError) object;
+
+                String field = fieldError.getField();
+                String errorMessage = fieldError.getDefaultMessage();
+
+                location = (field.replaceAll("(.)([A-Z])", "$1_$2")).toLowerCase();
+
+                if (errorMessage.equals("VALUE_OUTSIDE_RANGE")) {
```

Comment:
Can we write this line as `if ("VALUE_OUTSIDE_RANGE".equals(errorMessage)) {` to get a null check for free.